src/main.rs

```diff
@@ -200,82 +375,56 @@ async fn redirect(State(shared_state): State<AppState>, data_string: String) ->
         file_name: "wicked.xml".to_string(),
         file_type: FileType::Xml,
     }];
-    let path = match migrate(data_arr) {
-        Ok(path) => path,
-        Err(_e) => return StatusCode::INTERNAL_SERVER_ERROR.into_response(),
-    };
 
-    let uuid = match create_and_add_row(path, &database) {
+    let uuid = match migrate(data_arr, &database) {
         Ok(uuid) => uuid,
-        Err(_e) => return StatusCode::INTERNAL_SERVER_ERROR.into_response(),
+        Err(e) => return e.into_response(),
     };
-    axum::response::Redirect::to(format!("/{}", uuid).as_str()).into_response()
-}
 
-fn migrate(data_arr: Vec<File>) -> Result<String, anyhow::Error> {
-    let output_tmpfile: tempfile::NamedTempFile = tempfile::Builder::new()
-        .prefix("nm-migrated.")
-        .suffix(".tar")
-        .keep(true)
-        .tempfile()?;
-
-    let output_path_str: &str = match output_tmpfile.path().to_str() {
-        Some(output_path_str) => output_path_str,
-        None => return Err(anyhow::anyhow!("Failed to convert path to string")),
-    };
-
-    let migration_target_tmpdir: tempfile::TempDir = tempdir()?;
+    Redirect::to(&format!("/{}", uuid)).into_response()
+}
 
-    for file in &data_arr {
-        let input_file_path = migration_target_tmpdir.path().join(&file.file_name);
+fn create_and_write_to_file(
+    files: &Vec<File>,
+    migration_target_path: String,
+) -> Result<(), anyhow::Error> {
+    for file in files {
+        let input_file_path = migration_target_path.clone() + "/" + &file.file_name;
         fs::File::create_new(&input_file_path)?;
         std::fs::write(&input_file_path, file.file_content.as_bytes())?;
```

Comment:
The docs for `std::fs::write` mention:
```
This function will create a file if it does not exist, and will entirely replace its contents if it does.
```
So the `create_new` call can be removed.